mops/include/mops/checks.hpp

```diff
@@ -0,0 +1,74 @@
+#ifndef MOPS_CHECKS_HPP
+#define MOPS_CHECKS_HPP
+
+#include "mops/tensor.hpp"
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <stdexcept>
+#include <string>
+
+template <typename T_1, size_t N_DIMS_1, typename T_2, size_t N_DIMS_2>
```

Comment:
Maybe a quick doc comment on what the function does and what the template parameters are?